src/client/index.ts

```diff
@@ -374,7 +385,7 @@ class Client extends EventEmitter {
     }
     const nextPageRequest = { ...req, marker: resp.result.marker }
     // eslint-disable-next-line @typescript-eslint/consistent-type-assertions -- Necessary for overloading
-    return this.connection.request(nextPageRequest) as unknown as U
+    return this.request(nextPageRequest as any) as unknown as U
```

Comment:
The partial payment handling code is in the `this.request` function. By calling `this.request` instead of `connection.request`, we get this handling code in `requestNextPage` as well.